www/qserv/js/ToolsSql.js

```diff
@@ -0,0 +1,219 @@
+    <div class="col">
+      <p>This tool allows executing queries directly against the <span style="font-weight:bold;">MariaDB</span>
+        service of a <span style="font-weight:bold;">Qserv</span> worker. For queries which produce a result set
+        the result will be displayed as a table below the form. Errors will be also reported.
+        Please, <span style="font-weight:bold;">DO NOT</span> launch queries which may produce millions of rows.
+        An attept to do so will make thsi applicatio unresponsive.
+        Use <span style="font-style:italic;">SELECT COUNT(*)</span> first.
+      </p>
+    </div>
+  </div>
+  <div class="row">
+    <div class="col-3">
+      <input type="text" class="form-control" placeholder="Worker" id="worker">
+    </div>
+    <div class="col-3">
+      <input type="text" class="form-control" placeholder="User" id="user">
+    </div>
+    <div class="col-3">
+      <input type="password" class="form-control" placeholder="Password" id="password">
+    </div>
+    <div class="col-3">
+      <button class="btn btn-primary   btn-sm" id="execute">Execute</button>
+      <button class="btn btn-secondary btn-sm" id="reset"  >Reset form</button>
+    </div>
+  </div>
+  <div class="row">
+    <div class="col">
+      <textarea rows="3" class="form-control" placeholder="Query" id="query"></textarea>
+    </div>
+  </div>
+  <div class="row">
+    <div class="col">
+      <table class="table table-sm table-hover table-bordered" id="fwk-tools-sql-resultset">
+        <caption></caption>
+        <thead class="thead-light">
+        </thead>
+        <tbody></tbody>
+      </table>
+    </div>
+  </div>
+</div>`;
+            this.fwk_app_container.html(html);
+            this.fwk_app_container.find('button#execute').click(() => {
+                this._load();
+            });
+            this.fwk_app_container.find('button#reset').click(() => {
+                this._worker().val('');
+                this._user().val('');
+                this._password().val('');
+                this._query().val('');
+            });
+        }
+
+        // Inputs
+
+        _worker() {
+            if (this._worker_obj === undefined) {
+                this._worker_obj = this.fwk_app_container.find('input#worker');
+            }
+            return this._worker_obj;
+        }
+        _user() {
+            if (this._user_obj === undefined) {
+                this._user_obj = this.fwk_app_container.find('input#user');
+            }
+            return this._user_obj;
+        }
+        _password() {
+            if (this._password_obj === undefined) {
+                this._password_obj = this.fwk_app_container.find('input#password');
+            }
+            return this._password_obj;
+        }
+        _query() {
+            if (this._query_obj === undefined) {
+                this._query_obj = this.fwk_app_container.find('textarea#query');
+            }
+            return this._query_obj;
+        }
```

Comment:
I see a lot of repetition in these methods, maybe a chance to refactor a bit?

Reply:
Good point. I will have a look at it.